tests/helpers/test_template.py

```diff
@@ -7200,3 +7200,49 @@ def test_combine(hass: HomeAssistant) -> None:
 
     with pytest.raises(TemplateError, match="combine expected a dict, got str"):
         template.Template("{{ {'a': 1} | combine('not a dict') }}", hass).async_render()
+
+
+def test_sort_naturally(hass: HomeAssistant) -> None:
```

Comment:
Should add a test for floating point numbers.  Probably a few different tests.  One with increasing sig figs, decreasing sig figs, and/or the same sig figs.

Reply:
Now that I think about this, we might need to add an option for a float sort for the increasing/decreasing sig figs.  That will likely change the sort order when treated like ints.  Have it default to an `int` sort but give the option to do a `float` sort.